### PR TITLE
CPLAT-6708: Add stack trace to more logs

### PR DESCRIPTION
**Pull request by georgelesica-wf**
# [CPLAT-6708](https://jira.atl.workiva.net/browse/CPLAT-6708)
Issue Status

## Motivation
  
It was brought to our attention by @laurameckley-wk that it might be helpful to have stack traces attached to a few more of the logs emitted by w_module in order to help track down the root cause of the corresponding warnings.

## Changes
  
Add stack traces to logs in `LifecycleModule`.

#### Release Notes
  
Add additional context to logs emitted by `LifecycleModule`.

## Review
_[See CONTRIBUTING.md][contributing-review-types] for more details on review types (+1 / QA +1 / +10) and code review process._

  

Please review: 

### QA Checklist
- [ ] Tests were updated and provide good coverage of the changeset and other affected code
- [ ] Manual testing was performed if needed
    - [ ] Steps from PR author: 
        
      - [ ] CI passed
    - [ ] Anything falling under manual testing criteria [outlined in CONTRIBUTING.md][contributing-manual-testing]

## Merge Checklist
While we perform many automated checks before auto-merging, some manual checks are needed:
- [ ] A Client Platform member has reviewed these changes
- [ ] There are no unaddressed comments _- this check can be automated if reviewers use the "Request Changes" feature_
- [ ] _For release PRs -_ Version metadata in Rosie comment is correct

[contributing-review-types]: https://github.com/Workiva/w_module/blob/master/CONTRIBUTING.md#review-types
[contributing-manual-testing]: https://github.com/Workiva/w_module/blob/master/CONTRIBUTING.md#manual-testing-criteria

### CODE REVIEW

**Comment by aviary2-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://dev.workiva.net/docs/teams/information-security/aviary#configuring-aviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: lib/src/lifecycle_module.dart [hunk 1]
```diff
@@ -964,8 +970,10 @@ abstract class LifecycleModule extends SimpleModule with Disposable {
   /// A utility to logging LifecycleModule lifecycle events
   void _logLifecycleEvents(
       String logLabel, Stream<dynamic> lifecycleEventStream) {
-    listenToStream(lifecycleEventStream, (_) => _logger.finest(logLabel),
-        onError: (error) => _logger.warning('$logLabel error: $error'));
+    listenToStream(lifecycleEventStream,
+        (_) => _logger.finest(logLabel, null, StackTrace.current),
```
Comment (greglittlefield-wf) [inline lib/src/lifecycle_module.dart]: Does we know whether this significantly affects performance? There are a lot of lifecycle events that get logged during app initialization.

Comment (greglittlefield-wf) [reply lib/src/lifecycle_module.dart]: On warnings, which should happen much less frequently, I think it's fine; I'm just worried about normal logs

Comment (georgelesica-wf) [reply lib/src/lifecycle_module.dart]: Hmm, fair point. I probably got a little over-zealous with that one.

**Comment by laurameckley-wk** [comment]
+1

**Comment by corwinsheahan-wf** [comment]
QA +1
* Log updates only, CI Passing is sufficient.

Merging into master.
@Workiva/release-management-p

**Pull request merged**